### PR TITLE
[feat]: [CI-12484]: Add support for RunTestsV2

### PR DESCRIPTION
**Pull request by anurag-harness**
This PR adds supports for RunTestV2 in Harness Docker Runner. 

Testing Done: 

Java
Full Run: https://app.harness.io/ng/account/9UuUfLwaQ-6ZowvbG7qtLQ/module/ci/orgs/default/projects/anuragworkspace/pipelines/JHTTP_Linux_Docker_Runner/executions/_O9I1ZoQTr-JVGJg_8K_yQ/pipeline?storeType=INLINE
TI Run: https://app.harness.io/ng/account/9UuUfLwaQ-6ZowvbG7qtLQ/module/ci/orgs/default/projects/anuragworkspace/pipelines/JHTTP_Linux_Docker_Runner/executions/y8Z2gBDJTx2s1Az38isM9g/pipeline?storeType=INLINE

Made sure precmd and env variables injected properly, test selection happens, cg gets uploaded and test reports get uploaded by default

### CODE REVIEW

File: pipeline/runtime/step_executor.go [hunk 1]
```diff
@@ -307,6 +307,9 @@ func (e *StepExecutor) run(ctx context.Context, engine *engine.Engine, r *api.St
 	if r.Kind == api.Run {
 		return executeRunStep(ctx, engine, r, out, tiConfig)
 	}
+	if r.Kind == api.RunTestsV2 {
+		return executeRunTestsV2Step(ctx, engine, r, out, tiConfig)
+	}
```
Comment (smjt-h) [inline pipeline/runtime/step_executor.go:312]: Will RunTestV1 work here, I see there is no condition for that?

Comment (anurag-salesforce) [reply pipeline/runtime/step_executor.go:312]: Yeah if it is neither then the final return is executeRunTestStep. I have tested it and it works. This is the same as on lite-engine

Comment (ShobhitSingh11) [reply pipeline/runtime/step_executor.go:312]: I think we do have condition for V1 in LE 
Screenshot 2024-08-01 at 10 59 28 PM

Comment (anurag-salesforce) [reply pipeline/runtime/step_executor.go:312]: Yes it is the same here

**Review by ShobhitSingh11 — COMMENTED** [review]
Are we not supporting intelligence mode here like we are doing in LE ??

**Comment by anurag-harness** [comment]
> Are we not supporting intelligence mode here like we are doing in LE ??

We are, it is being handled inside SetupRunTestV2 in lite-engine which we have imported and used here: https://github.com/harness/lite-engine/blob/b3e2501d28abd53142b0ed23146c8d3b1755f09d/pipeline/runtime/runtestsV2.go#L140

**Review by smjt-h — APPROVED** [review]

**Pull request merged**